datafusion/physical-plan/src/statistics.rs

```diff
@@ -131,14 +95,35 @@ impl MinMaxStatistics {
                 .collect::<Vec<_>>(),
         };
 
+        let projected_statistics = statistics
+            .into_iter()
+            .cloned()
+            .map(|s| s.project(Some(&projection)))
+            .collect::<Vec<_>>();
+
+        // Helper function to get min/max statistics
+        let get_min_max = |i: usize| -> Result<(Vec<ScalarValue>, Vec<ScalarValue>)> {
+            Ok(projected_statistics
+                .iter()
```

Comment:
I moved this code later so it uses the _projected_ statistics, i.e. it only relies on stats for sorting columns. I was seeing the code error because some columns had unknown statistics.